src/database/sqlite/mod.rs

```diff
@@ -889,6 +912,315 @@ CREATE TABLE spend_transactions (
+                && coin[0].outpoint == coin_b.outpoint
+                && coin[1].outpoint == coin_c.outpoint
+                && coin[2].outpoint == coin_d.outpoint));
+
+            // We can also get two of the three unconfirmed coins by filtering for their outpoints.
+            assert!(vec![
+                conn.coins(&[CoinStatus::Unconfirmed], &[outpoint_b, outpoint_c]),
+                conn.coins(&[], &[outpoint_b, outpoint_c]),
+                conn.db_coins(&[outpoint_b, outpoint_c]),
+            ]
+            .iter()
+            .all(|coin| coin.len() == 2
+                && coin[0].outpoint == coin_b.outpoint
+                && coin[1].outpoint == coin_c.outpoint));
+
+            // Now spend second coin, even though it is still unconfirmed.
+            conn.spend_coins(&[(
+                coin_b.outpoint,
+                bitcoin::Txid::from_slice(&[1; 32][..]).unwrap(),
+            )]);
+            // The coin shows as spending.
+            assert!(vec![
+                conn.coins(&[CoinStatus::Spending], &[]),
+                conn.coins(&[CoinStatus::Spending], &[outpoint_b]),
+                conn.coins(&[], &[outpoint_b]),
+                conn.list_spending_coins(),
+                conn.db_coins(&[outpoint_b])
+            ]
+            .iter()
+            .all(|res| res.len() == 1 && res[0].outpoint == coin_b.outpoint));
+
+            // Now confirm the third coin.
+            conn.confirm_coins(&[(coin_c.outpoint, 175500, 175500)]);
+
+            // We now only have one unconfirmed coin.
+            assert!(vec![
+                conn.coins(&[CoinStatus::Unconfirmed], &[]),
+                conn.coins(
+                    &[CoinStatus::Unconfirmed],
+                    &[outpoint_a, outpoint_b, outpoint_c, outpoint_d]
+                ),
+                conn.coins(&[], &[outpoint_d]),
+                conn.db_coins(&[outpoint_d]),
+            ]
+            .iter()
+            .all(|c| c.len() == 1 && c[0].outpoint == coin_d.outpoint));
+
+            // There is now one coin for each status.
+            assert!(vec![
+                conn.coins(&[CoinStatus::Unconfirmed], &[]),
+                conn.coins(&[CoinStatus::Unconfirmed], &[outpoint_d]),
+                conn.coins(&[CoinStatus::Confirmed], &[]),
+                conn.coins(&[CoinStatus::Confirmed], &[outpoint_c]),
+                conn.coins(&[CoinStatus::Spending], &[]),
+                conn.coins(&[CoinStatus::Spending], &[outpoint_b]),
+                conn.coins(&[CoinStatus::Spent], &[]),
+                conn.coins(&[CoinStatus::Spent], &[outpoint_a]),
+                conn.coins(&[], &[outpoint_a]),
+                conn.coins(&[], &[outpoint_b]),
+                conn.coins(&[], &[outpoint_c]),
+                conn.coins(&[], &[outpoint_d]),
+            ]
+            .iter()
+            .map(|c| c.len())
+            .all(|length| length == 1));
+
+            fs::remove_dir_all(tmp_dir).unwrap();
+        }
```

Comment:
This ought to be outside the scope of the connection object.

```suggestion
        }
    fs::remove_dir_all(tmp_dir).unwrap();
```

That'll fix the Windows CI.

Reply:
Thanks for this! The failing Windows test is now passing :relieved: